graylog2-web-interface/src/views/components/visualizations/worldmap/WorldMapVisualization.jsx

```diff
@@ -1,37 +1,48 @@
 // @flow strict
 import React, { useContext } from 'react';
 import PropTypes from 'prop-types';
-import { flow, fromPairs, get, zip } from 'lodash';
+import { flow, fromPairs, get, zip, isEmpty } from 'lodash';
 
 import { AggregationType } from 'views/components/aggregationbuilder/AggregationBuilderPropTypes';
 import WorldMapVisualizationConfig from 'views/logic/aggregationbuilder/visualizations/WorldMapVisualizationConfig';
 import Viewport from 'views/logic/aggregationbuilder/visualizations/Viewport';
 import type { VisualizationComponent, VisualizationComponentProps } from 'views/components/aggregationbuilder/AggregationBuilder';
+import type { Rows } from 'views/logic/searchtypes/pivot/PivotHandler';
 
 import MapVisualization from './MapVisualization';
-import { extractSeries, formatSeries } from '../ChartData';
+import { extractSeries, formatSeries, getLeafsFromRows, getXLabelsFromLeafs } from '../ChartData';
 import transformKeys from '../TransformKeys';
 import RenderCompletionCallback from '../../widgets/RenderCompletionCallback';
 
-const arrayToMap = ([name, x, y]) => ({ name, x, y });
-const lastKey = keys => keys[keys.length - 1];
-const mergeObject = (prev, last) => Object.assign({}, prev, last);
+const _arrayToMap = ([name, x, y]) => ({ name, x, y });
+const _lastKey = keys => keys[keys.length - 1];
+const _mergeObject = (prev, last) => Object.assign({}, prev, last);
+const _createSeriesWithoutMetric = (rows: Rows) => {
+  const leafs = getLeafsFromRows(rows);
+  const xLabels = getXLabelsFromLeafs(leafs);
+  return { valuesBySeries: { 'No metric defined': xLabels.map(() => null) }, xLabels };
+};
 
 const WorldMapVisualization: VisualizationComponent = ({ config, data, editing, onChange, width, ...rest }: VisualizationComponentProps) => {
-  const onRenderComplete = useContext(RenderCompletionCallback);
   const { rowPivots } = config;
+  const onRenderComplete = useContext(RenderCompletionCallback);
+  const hasMetric = !isEmpty(config._value.series);
+  const markerRadiusSize = !hasMetric ? 1 : undefined;
+  const seriesExtractor = hasMetric ? extractSeries() : _createSeriesWithoutMetric;
+
   const pipeline = flow([
     transformKeys(config.rowPivots, config.columnPivots),
-    extractSeries(),
+    seriesExtractor,
     formatSeries,
-    results => results.map(arrayToMap),
+    results => results.map(_arrayToMap),
   ]);
+
   const series = pipeline(data)
     .map(({ name, x, y }) => {
-      const newX = x.map(lastKey);
+      const newX = x.map(_lastKey);
```

Comment:
Optional: Not an issue with your code, but while you are working on it, I think it is a good idea to extract these two `.map(...)` calls into separate functions and them to the `flow([...])` statement, as they are part of the pipeline.

Reply:
Good idea. I've created another function and also summarized the two `.map`s in one `.map`.